cmake/Reprojucer.cmake

```diff
@@ -345,19 +351,26 @@ function(jucer_project_module module_name PATH_TAG modules_folder)
   foreach(src_file ${module_src_files})
     unset(to_compile)
 
-    # See LibraryModule::CompileUnit::isNeededForExporter()
-    # in JUCE/extras/Projucer/Source/Project/jucer_Module.cpp
-    if(  (src_file MATCHES "_AU[._]"         AND NOT (JUCER_BUILD_AUDIOUNIT    AND APPLE))
-      OR (src_file MATCHES "_AUv3[._]"       AND NOT (JUCER_BUILD_AUDIOUNIT_V3 AND APPLE))
-      OR (src_file MATCHES "_AAX[._]"        AND NOT (JUCER_BUILD_AAX          AND (APPLE OR MSVC)))
-      OR (src_file MATCHES "_RTAS[._]"       AND NOT (JUCER_BUILD_RTAS         AND (APPLE OR MSVC)))
-      OR (src_file MATCHES "_VST2[._]"       AND NOT (JUCER_BUILD_VST          AND TRUE))
-      OR (src_file MATCHES "_VST3[._]"       AND NOT (JUCER_BUILD_VST3         AND (APPLE OR MSVC)))
-      OR (src_file MATCHES "_Standalone[._]" AND NOT (JUCER_BUILD_AUDIOUNIT_V3 AND APPLE))
+    if(  (src_file MATCHES "_AU[._]"   AND NOT (JUCER_BUILD_AUDIOUNIT    AND APPLE))
+      OR (src_file MATCHES "_AUv3[._]" AND NOT (JUCER_BUILD_AUDIOUNIT_V3 AND APPLE))
+      OR (src_file MATCHES "_AAX[._]"  AND NOT (JUCER_BUILD_AAX          AND (APPLE OR MSVC)))
+      OR (src_file MATCHES "_RTAS[._]" AND NOT (JUCER_BUILD_RTAS         AND (APPLE OR MSVC)))
+      OR (src_file MATCHES "_VST2[._]" AND NOT (JUCER_BUILD_VST          AND TRUE))
+      OR (src_file MATCHES "_VST3[._]" AND NOT (JUCER_BUILD_VST3         AND (APPLE OR MSVC)))
     )
       set(to_compile FALSE)
     endif()
 
+    if(DEFINED JUCER_VERSION AND JUCER_VERSION VERSION_LESS 5)
+      if(src_file MATCHES "_Standalone[._]" AND NOT (JUCER_BUILD_AUDIOUNIT_V3 AND APPLE))
+        set(to_compile FALSE)
+      endif()
+    else()
+      if(src_file MATCHES "_Standalone[._]" AND NOT JUCER_BUILD_STANDALONE_PLUGIN)
```

Comment:
You could combine the two `if` statements into `(!(audio_unit_v3 && APPLE) || !(jucer_build_standalone_plugin)`

Reply:
I cannot do that, since the condition on `JUCER_VERSION` matters. In JUCE 4, files that match `_Standalone[._]` should be compiled only when the user has enabled `Build AudioUnit v3`, but in JUCE 5, these files should be compiled only when the user has enabled `Build Standalone Plug-In`.

Reply:
Ah yes, I see it now..